elasticsearch/src/main/java/org/apache/calcite/adapter/elasticsearch/ElasticsearchRules.java

```diff
@@ -113,8 +113,7 @@ static List<String> elasticsearchFieldNames(final RelDataType rowType) {
     return SqlValidatorUtil.uniquify(
         new AbstractList<String>() {
           @Override public String get(int index) {
-            final String name = rowType.getFieldList().get(index).getName();
-            return name.startsWith("$") ? "_" + name.substring(2) : name;
+            return rowType.getFieldList().get(index).getName();
```

Comment:
Before tearing down this chesterton fence... do we know what was the logic behind this piece of code that you propose to remove?

Reply:
I think it's not works. This function similar to org.apache.calcite.adapter.mongodb.MongoRules#mongoFieldNames which commit-id is ce0eb9c4.

Reply:
But elasticsearch adapter is different, In my above description,  when join condition push  down to project, _MAP['doc_id'] field name is "$f1",  correspond to "doc_id" field in _source of elasticsearch index. So the value of  fields is ["$f1"],  mappings values is  {"$f1": "doc_id"}, fields and mappings are paramter of   ElasticsearchTable.ElasticsearchQueryable#find.

Function elasticsearchFieldNames called by ElasticsearchToEnumerableConverter transform "$f1" of fields to "_1"  but dose not transform key of mappings. so column value of "$f1" is null.